### DISCUSSION TITLE
Create repository using template

### DISCUSSION THREAD

**Discussion by eypigu** [community/community#24409] [Apps, API and Webhooks]
<p>Hello there!</p>
<p>I was wondering if there was any way to create a repository using the new introduced feature of templates from the API, I couldn’t find anything related to it.</p>
<p>If it’s not yet implemented, what are the plans for this feature to be supported in the API?</p>

**Comment by jits** [comment]
<p>A big +1 for this.</p>
<p>It’d be useful to be able to:</p>
<ol>
<li>List all repos in an org that have the template flag set.</li>
<li>Specify a repo as the template for the create repository API.</li>
<li>Have a reference to the original template repo when retrieving metadata about a repository.</li>
</ol>

**Comment by AndreaGriffiths11** [comment, marked as answer]
<p>Hi <a class="mention" href="/undervane">@undervane</a> &amp; <a class="mention" href="/jits">@jits</a> ,</p>
<p>Thanks for this feedback! We’re always working to improve GitHub and the GitHub Community Forum, and we consider every suggestion we receive. I’ve logged your feature request in our internal feature request list.</p>
<p>Though I can’t guarantee anything or share a timeline for this, I can tell you that it’s been shared with the appropriate teams for consideration.</p>
<p>Please let me know if you have any other questions.</p>
<p>Cheers!</p>

**Comment by gleichda** [comment]
<p>For me this would be a basic need. Can you tell me as soon as you are having an ETA for this task please :slight_smile:</p>
